File: solutions/fortran/leap/1/leap.f90
module leap
  implicit none

contains

  logical function is_leap_year(year)
    integer :: year
    if (mod(year,4)==0 .and. &
       (mod(year,100) /=0 .or. &
        mod(year,400) == 0)) then
      is_leap_year = .true.
    else 
      is_leap_year = .false.
    end if
  end function

end module

File: solutions/r/leap/1/leap.R
leap <- function(year) {
	if (year %% 400 == 0) {
		return (TRUE)
	} else if (year %% 100 == 0) {
		return (FALSE)
	} else if (year %% 4 == 0) {
		return (TRUE)
	}
	return (FALSE)		
}
